internal/testutils/xds/e2e/clientresources.go

```diff
@@ -159,14 +162,54 @@ func DefaultServerListener(host string, port uint32, secLevel SecurityLevel) *v3
 	}
 
 	var ts *v3corepb.TransportSocket
+
 	if tlsContext != nil {
 		ts = &v3corepb.TransportSocket{
 			Name: "envoy.transport_sockets.tls",
 			ConfigType: &v3corepb.TransportSocket_TypedConfig{
-				TypedConfig: testutils.MarshalAny(tlsContext),
+				TypedConfig: testutils.MarshalAny(&testing.T{}, tlsContext),
 			},
 		}
 	}
+
+	filterTypedConfig := testutils.MarshalAny(&testing.T{}, &v3httppb.HttpConnectionManager{
```

Comment:
Is there any difference between `filterTypedConfig` and `filteredTypedConfig`? Also passing an empty `testing.T` is not what we want.

Reply:
It might be too painful to convert functions in this package to accept a `testing.T` that can be forwarded along `testutils.MarshalAny`. Instead you could have an unexported function in this function which resembles `panic`s instead of calling `t.Fatal` (like the old implementation).